apps/web/src/views/Predictions/hooks/usePredictionPrice.ts

```diff
@@ -23,10 +22,40 @@ interface PriceResponse {
   currencyB: PriceApiWhitelistedCurrency | (string & NonNullable<unknown>)
 }
 
+const DEFAULT_CURRENCY_A: PriceApiWhitelistedCurrency = 'ETH'
+const DEFAULT_CURRENCY_B: PriceApiWhitelistedCurrency = 'USDT'
+const DEFAULT_POLLING_INTERVAL = 5_000
+
+/**
+ * Mutate the price directly from the source API if
+ * Live Price is needed urgently
+ */
+export const usePredictionPriceMutation = () => {
+  const queryClient = useQueryClient()
+
+  return useMutation({
```

Comment:
@memoyil The mutation function will call Binance API directly to go around our own APIs caching. 
This data is set in query data and cached so will be reflected everywhere on the UI where the same `queryKey` exists

Reply:
True, it is mainly for after Updates are made from server and we already have data. 

But I want to use a different data source and `refetch` will use the existing one.

How would you recommend I implement a different data source for a one-off call. I had considered passing param like `fetchFromSource?: boolean` but the consumer of the hook would need to manage state for once-in-a-while use

Reply:
I want to use a different data source and refetch function would use the existing api.

I have updated to make the update function a normal async call instead of a mutation